storage/namespace.go

```diff
@@ -505,12 +540,19 @@ func (n *dbNamespace) Flush(
 		return nil
 	}
 
+	// check if blockStart is aligned with the namespace's retention options
+	bs := n.nopts.RetentionOptions().BlockSize()
+	truncated := blockStart.Truncate(bs)
+	if truncated != blockStart {
```

Comment:
Need to use `a.Equal(b)` when comparing time on account of the TZ stuff.  Also this is a pretty major nit on the style, but in future can put this on a single line `if t := blockStart.Truncate(bs); !t.Equal(blockStart) { ... }`